### PR TITLE
[SPARK-29664][PYTHON][SQL] Column.getItem behavior is not consistent with Scala

### PR DESCRIPTION
**Pull request by imback82**
### What changes were proposed in this pull request?

This PR changes the behavior of `Column.getItem` to call `Column.getItem` on Scala side instead of `Column.apply`.

### Why are the changes needed?

The current behavior is not consistent with that of Scala.

In PySpark:
```Python
df = spark.range(2)
map_col = create_map(lit(0), lit(100), lit(1), lit(200))
df.withColumn("mapped", map_col.getItem(col('id'))).show()
# +---+------+
# | id|mapped|
# +---+------+
# |  0|   100|
# |  1|   200|
# +---+------+
```
In Scala:
```Scala
val df = spark.range(2)
val map_col = map(lit(0), lit(100), lit(1), lit(200))
// The following getItem results in the following exception, which is the right behavior:
// java.lang.RuntimeException: Unsupported literal type class org.apache.spark.sql.Column id
//  at org.apache.spark.sql.catalyst.expressions.Literal$.apply(literals.scala:78)
//  at org.apache.spark.sql.Column.getItem(Column.scala:856)
//  ... 49 elided
df.withColumn("mapped", map_col.getItem(col("id"))).show
```

### Does this PR introduce any user-facing change?

Yes. If the use wants to pass `Column` object to `getItem`, he/she now needs to use the indexing operator to achieve the previous behavior.

```Python
df = spark.range(2)
map_col = create_map(lit(0), lit(100), lit(1), lit(200))
df.withColumn("mapped", map_col[col('id'))].show()
# +---+------+
# | id|mapped|
# +---+------+
# |  0|   100|
# |  1|   200|
# +---+------+
```

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by imback82** [comment]
cc: @HyukjinKwon

**Review by viirya — COMMENTED** [review]
To be consistent sounds ok. A concern is about this is a behavior change could break user code.

**Comment by HyukjinKwon** [comment]
Yeah, true. Workaround is pretty easy here (`df[col]`) instead. I think it should be fine since we're in Spark 3.

File: python/pyspark/sql/tests/test_column.py [hunk 1]
```diff
@@ -23,6 +23,8 @@
 from pyspark.sql.utils import AnalysisException
 from pyspark.testing.sqlutils import ReusedSQLTestCase
 
+from py4j.protocol import Py4JJavaError
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_column.py]: Can you move `py4j` over `pyspark` per pep8 (https://www.python.org/dev/peps/pep-0008/#imports)?

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise.

File: python/pyspark/sql/tests/test_column.py (hunk 1 above)
Comment (imback82) [reply python/pyspark/sql/tests/test_column.py]: Fixed. Thanks!

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**